xtra-libp2p-rollover/src/v_2_0_0/maker.rs

```diff
@@ -0,0 +1,395 @@
+    R: Send + Sync + 'static,
+{
+    type Stop = ();
+
+    async fn stopped(self) -> Self::Stop {}
+}
+
+#[xtra_productivity]
+impl<E, O, R> Actor<E, O, R>
+where
+    E: ExecuteOnCfd + Clone + Send + Sync + 'static,
+    O: GetAnnouncements + Clone + Send + Sync + 'static,
+    R: GetRates + Clone + Send + Sync + 'static,
+{
+    async fn handle(&mut self, msg: UpdateConfiguration) {
+        self.is_accepting_rollovers = msg.is_accepting_rollovers;
+    }
+}
+
+#[xtra_productivity]
+impl<E, O, R> Actor<E, O, R>
+where
+    E: ExecuteOnCfd + Clone + Send + Sync + 'static,
+    O: GetAnnouncements + Clone + Send + Sync + 'static,
+    R: GetRates + Clone + Send + Sync + 'static,
+{
+    async fn handle(&mut self, msg: NewInboundSubstream, ctx: &mut xtra::Context<Self>) {
+        let NewInboundSubstream { peer, stream } = msg;
+        let address = ctx.address().expect("we are alive");
+
+        tokio_extras::spawn_fallible(
+            &address.clone(),
+            async move {
+                let mut framed =
+                    Framed::new(stream, JsonCodec::<ListenerMessage, DialerMessage>::new());
+
+                let propose = framed
+                    .next()
+                    .await
+                    .context("End of stream while receiving Propose")?
+                    .context("Failed to decode Propose")?
+                    .into_propose()?;
+
+                address
+                    .send(ProposeReceived {
+                        propose,
+                        framed,
+                        peer,
+                    })
+                    .await?;
+
+                anyhow::Ok(())
+            },
+            move |e| async move {
+                tracing::warn!(%peer, "Failed to handle incoming rollover protocol: {e:#}")
+            },
+        );
+    }
+
+    async fn handle(&mut self, msg: ProposeReceived) {
+        let ProposeReceived {
+            propose,
+            mut framed,
+            peer,
+        } = msg;
+        let order_id = propose.order_id;
+
+        let (from_event_id, from_complete_fee) = match self
+            .executor
+            .execute(order_id, |cfd| {
+                cfd.verify_counterparty_peer_id(&peer.into())?;
+                cfd.start_rollover_maker(propose.from_commit_txid)
+            })
+            .await
+        {
+            Ok(event_id) => event_id,
+            Err(e) => {
+                tracing::warn!(%order_id, "Rollover failed after handling taker proposal: {e:#}");
+
+                // We have to append failed to ensure that we can rollover in the future
+                // The cfd logic might otherwise prevent us from starting a rollover if there is
+                // still one ongoing that was not properly ended.
+                emit_failed(order_id, e, &self.executor).await;
+
+                return;
+            }
+        };
+
+        if !self.is_accepting_rollovers {
+            emit_rejected(order_id, &self.executor).await;
```

Comment:
So, there is no `accept` anymore, but there is a `reject`?
Does that mean that the taker could either receive a `reject` or the first rollover message in the protocol flow?

Reply:
Sorry, my bad, there is still an accept